Template updates for Aspire 9.4 and .NET 10

DamianEdwards opened the pull request:
## Description

Updates the templates for Aspire 9.4 and .NET 10.

Fixes #6764

![Screenshot 2025-06-13 172511](https://github.com/user-attachments/assets/e3da1d63-74d8-4163-8cef-53153d1d140f)

<img width="1517" alt="image" src="https://github.com/user-attachments/assets/6528a36a-b3f4-45c4-aee7-cf8f1cc3dc55" />


## Checklist

- Is this feature complete?
  - [ ] Yes. Ready to ship.
  - [x] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
- Did you add public API?
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [x] No
- Does the change require an update in our Aspire docs?
  - [x] Yes
    - Link to aspire-docs issue (consider using one of the following templates): dotnet/docs-aspire#3820

DamianEdwards committed:
Update apphost template for 9.4 and net10.0

DamianEdwards committed:
Update templates for Aspire 9.4 & .NET 10

DamianEdwards committed:
Rollback xunit.runner.visualstudio to 3.1.0

DamianEdwards committed:
Rollback update to NUnit.Analyzers version

mitchdenny commented:
Would be good to update the devcontainer definition as well to have .NET 10, 9 and 8 installed.

joperezr reviewed (approved):
Thanks for working on this @DamianEdwards, this looks good to me. Apart from the changes you added in the tests due to the new SDK this time around, how manual are these updates? Is there anything here that can/should be automated?

DamianEdwards commented:
@joperezr I used GHCP agent in VSCode to automate some of this after I updated the first template, but after a bit of iteration it still missed some things so I completed it manually. This specific update was more complex than usual though as we added support for another TFM, vs. just updating for an Aspire point release.

I think next step for automation attempt would be for me to manually document the steps performed for a regular point release update. Once that's captured it should be a better place to start from to attempt automating this.

DamianEdwards committed:
Update test targets for new TFM

DamianEdwards committed:
Merge branch 'main' into damianedwards/templates-net10-support

DamianEdwards committed:
Add BuidEnvironment for ForNextSdkAndCurrentRuntime

radical commented:
Test changes look good. I didn't get a chance to look at anything else.
